src/Timecode.php

```diff
@@ -0,0 +1,413 @@
+        $this->dropFrame = is_null($dropFrame) ? self::$defaultDropFrame : $dropFrame;
+
+        if (! Validations::isValidTimeCode($time, $this->frameRate, $this->dropFrame)) {
+            throw new \InvalidArgumentException('Invalid timecode');
+        }
+
+        $this->generateFrameCount($time);
+    }
+
+    /**
+     * Convert the Timecode to its string representation.
+     *
+     * @return string
+     */
+    public function __toString() : string
+    {
+        return $this->toString();
+    }
+
+    /**
+     * Calculate the frame count based on time code
+     *
+     * @param string $time
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return int
+     */
+    public static function frameCountFromTimecode($time, $frameRate = null, $dropFrame = null) : int
+    {
+        $frameRate = is_null($frameRate) ? self::$defaultFrameRate : $frameRate;
+        $dropFrame = is_null($dropFrame) ? self::$defaultDropFrame : $dropFrame;
+
+        if (! Validations::isValidString($time, $frameRate, $dropFrame)) {
+            throw new \InvalidArgumentException('Invalid string format');
+        }
+
+        $times = preg_split('/(:|;)/', $time);
+        list($hours, $minutes, $seconds, $frames) = array_map(function ($time) {
+            return (int) $time;
+        }, $times);
+
+        $roundFrameRate = round($frameRate);
+        $frameCount = self::calculateFrameCount($roundFrameRate, $hours, $minutes, $seconds, $frames);
+
+        if ($dropFrame) {
+            $totalMinutes = (60 * $hours) + $minutes;
+
+            return (int) ($frameCount - (2 * ($totalMinutes - floor($totalMinutes / 10))));
+        }
+
+        return (int) $frameCount;
+    }
+
+    /**
+     * Create a new Timecode of a given seconds
+     *
+     * @param int $seconds
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return self
+     */
+    public static function fromSeconds($seconds, $frameRate = null) : self
+    {
+        $frameRate = is_null($frameRate) ? self::$defaultFrameRate : $frameRate;
+
+        if (! is_numeric($seconds)) {
+            throw new \InvalidArgumentException('First argument must be a number');
+        }
+
+        return new self(intval($seconds * $frameRate), $frameRate);
+    }
+
+    /**
+     * Calculate frame count
+     *
+     * @param float $frameRate
+     * @param int $hours
+     * @param int $minutes
+     * @param int $seconds
+     * @param int $frames
+     * @return float
+     */
+    public static function calculateFrameCount(
+        float $frameRate,
+        int $hours,
+        int $minutes,
+        int $seconds,
+        int $frames
+    ) : float {
+        return ($frameRate * 60 * 60 * $hours)
+            + ($frameRate * 60 * $minutes)
+            + ($frameRate * $seconds)
+            + $frames;
+    }
+
+    /**
+     * @param float $frameRate
+     */
+    public static function setDefaultFrameRate(float $frameRate)
```

Comment:
`Return type`

```suggestion
    public static function setDefaultFrameRate(float $frameRate) : void
```

Reply:
@wesleyhf the `PHPUnit` understands that `void` is a `FireworkWeb\SMPTE\void` instance, so we get a bug